File: src/Scripting/Ruby/patterns/client_server.rb
# frozen_string_literal: true

module RubyPatterns
  module ClientServer
    module_function

    def run
      server = ->(request) { { echo: request } }
      client = ->(value) { server.call(value)[:echo] }
      raise 'Client-Server failed' unless client.call('ping') == 'ping'
      true
    end
  end
end

RubyPatterns::ClientServer.run if $PROGRAM_NAME == __FILE__

File: src/Scripting/PHP/patterns/client_server.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('client_server');
$server = static fn(array $request): array => ['status'=>200,'body'=>strtoupper($request['body'])];
$client = static fn(string $body): array => $server(['body'=>$body]);
$check($client('ping')['body'] === 'PING');
